### PR TITLE
Implement flush for cloud logging handlers

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

File: logging/google/cloud/logging/handlers/transports/base.py [hunk 1]
```diff
@@ -33,3 +33,10 @@ def send(self, record, message):
                         formatted by the associated log formatters.
         """
         raise NotImplementedError
+
+    def flush(self):
+        """Submit any pending log records.
+
+        For blocking/sync transports, this is a no-op.
+        """
+        pass
```
Comment (dhermes) [inline logging/google/cloud/logging/handlers/transports/base.py]: `pass` not needed if there is a docstring (magic!)

File: logging/tests/unit/handlers/transports/test_background_thread.py [hunk 2]
```diff
@@ -233,6 +243,12 @@ def test__thread_main_batches(self):
         self.assertFalse(worker._cloud_logger._batch.commit_called)
         self.assertEqual(worker._queue.qsize(), 0)
 
+    def test_flush(self):
+        worker = self._make_one(_Logger(self.NAME))
+
+        # Queue is empty, should not block.
+        worker.flush()
```
Comment (dhermes) [inline logging/tests/unit/handlers/transports/test_background_thread.py]: This test needs some assertions

File: logging/tests/unit/handlers/transports/test_base.py [hunk 3]
```diff
@@ -32,3 +32,7 @@ def test_send_is_abstract(self):
         target = self._make_one()
         with self.assertRaises(NotImplementedError):
             target.send(None, None)
+
+    def test_flush_is_abstract_and_optional(self):
+        target = self._make_one()
+        target.flush()
```
Comment (dhermes) [inline logging/tests/unit/handlers/transports/test_base.py]: Probably can't assert much here. `ret_val is None`?

File: logging/google/cloud/logging/handlers/transports/base.py (hunk 1 above)
Comment (jonparrott) [inline logging/google/cloud/logging/handlers/transports/base.py]: Done.

File: logging/tests/unit/handlers/transports/test_background_thread.py (hunk 2 above)
Comment (jonparrott) [inline logging/tests/unit/handlers/transports/test_background_thread.py]: Done.

File: logging/tests/unit/handlers/transports/test_base.py (hunk 3 above)
Comment (jonparrott) [inline logging/tests/unit/handlers/transports/test_base.py]: Eh, I'd rather not.

**Comment by dhermes** [comment]
LGTM pending CI

**Pull request merged**